### PR TITLE
sets up artist data

### PR DESCRIPTION
**Pull request by rosscreighton**
- hooks up artist collection to album collection
- updates databaseSeeder to handle creation of a related artist when we seed an album
- sets up redux stuff for setting the artist in the store after we subscribe to the album publication

### CODE REVIEW

File: imports/api/Artists/server/publications.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { Artists } from '../Artists';
+
+Meteor.publish('artist', (id) => Artists.find({ _id: id }));
```
Comment (rosscreighton) [inline imports/api/Artists/server/publications.js]: i set up this publication but i didn't actually need it yet. see comment i left on albums publication.

File: imports/api/Albums/server/publications.js [hunk 2]
```diff
@@ -1,3 +1,9 @@
 import { Albums } from '../Albums';
```
Comment (rosscreighton) [inline imports/api/Albums/server/publications.js]: @sameckmeier take a look at what's happening here and in AlbumPageLoader. let me know if you don't follow why i did this and i'll explain it over the phone. i think we may want to consider a small design change to meteor loader so we don't have to do things like this.

File: imports/api/artists/Artists.js [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 
-const Artists = new Mongo.Collection('Artists');
+export const Artists = new Mongo.Collection('Artists');
 
 Artists.schema = new SimpleSchema({
   name: { type: String },
   genre: { type: Number },
   spotifyId: { type: String },
-  spotifySlug: { type: String },
+  slug: { type: String },
```
Comment (sameckmeier) [inline imports/api/artists/Artists.js]: 👍

File: imports/lib/database.js [hunk 4]
```diff
@@ -8,16 +10,45 @@ const documentsToSeed = [
...
+        newAttributes[attribute] = relatedId;
+      }
+    }
+  }
+
+  return newAttributes;
+}
+
 export function seedDatabase() {
+  resetDatabase();
```
Comment (sameckmeier) [inline imports/lib/database.js]: 👍

File: imports/redux/reducers.js [hunk 5]
```diff
@@ -27,8 +27,14 @@ export const defaultReviewsState = [
   },
 ];
 
-const artist = (state = {}, action) => {
-  return state;
+const artist = (state = { name: 'NAO'}, action) => {
+  let newState = state;
+
+  if (action.type === SET_ARTIST) {
```
Comment (sameckmeier) [inline imports/redux/reducers.js]: we'll also need to setup a spotify artist action/reducer

File: imports/api/Albums/server/publications.js (hunk 2 above)
Comment (sameckmeier) [inline imports/api/Albums/server/publications.js]: just to make sure, we covered this over the phone per our conversation about changes to MeteorLoader, ie removing Subscriber, so we can change this back to only returning an album?

File: .eslintrc [hunk 6]
```diff
@@ -12,3 +12,7 @@ globals:
 rules:
   import/no-unresolved: off
   no-underscore-dangle: off
+
+parserOptions:
+  ecmaFeatures:
+    experimentalObjectRestSpread: true
```
Comment (sameckmeier) [inline .eslintrc]: 👍

**Comment by sameckmeier** [comment]
@rosscreighton looks good! the only blocker is that we'll want to remove the patch for the album pub per our conversation on Sunday.

File: imports/api/Albums/server/publications.js (hunk 2 above)
Comment (rosscreighton) [inline imports/api/Albums/server/publications.js]: for sure, and i'll push the changes for this once i get the tests working 👍

**Comment by rosscreighton** [comment]
@sameckmeier alright its all going to master!!

**Pull request merged**